File: produto.java
import java.util.ArrayList;  // Importa a classe ArrayList para armazenar produtos dinamicamente
import java.util.Scanner;    // Importa a classe Scanner para entrada de dados do usuário
import java.util.Locale;     // Importa a classe Locale para configurar o formato numérico (ponto decimal)

// Classe que representa um produto com nome e preço
class Produto {
    String nome;      // Nome do produto (até 30 caracteres, mas em Java o limite é lógico, não fixo)
    float preco;      // Preço do produto

    // Construtor da classe Produto
    Produto(String nome, float preco) {
        this.nome = nome;
        this.preco = preco;
    }
}

public class CadastroProdutos {

    public static void main(String[] args) {
        Locale.setDefault(Locale.US); // Define o ponto como separador decimal
        Scanner scanner = new Scanner(System.in); // Cria scanner para entrada de dados

        ArrayList<Produto> produtos = new ArrayList<>(); // Lista dinâmica para armazenar produtos
        char continuar;

        do {
            // Solicita dados do produto ao usuário
            System.out.println("Produto " + (produtos.size() + 1)); // Exibe o número do produto atual

            System.out.print("Nome: ");
            String nome = scanner.nextLine(); // Lê o nome do produto

            System.out.print("Preço: ");
            float preco = scanner.nextFloat(); // Lê o preço do produto
            scanner.nextLine(); // Limpa o buffer após ler float

            // Cria novo objeto Produto e adiciona à lista
            produtos.add(new Produto(nome, preco));

            // Pergunta ao usuário se deseja adicionar outro produto
            System.out.print("Deseja adicionar outro produto? (s/n): ");
            continuar = scanner.nextLine().charAt(0); // Lê o primeiro caractere da resposta

        } while (continuar == 's' || continuar == 'S'); // Continua se usuário digitar 's' ou 'S'

        // Exibe os produtos cadastrados
        System.out.println("\nLista de Produtos:");
        for (Produto p : produtos) {
            System.out.printf("Produto: %s - Preço: R$ %.2f%n", p.nome, p.preco);
        }

        scanner.close(); // Fecha o Scanner (boa prática)
    }
}


// Produto 1
// Nome: Caneta Azul
// Preço: 2.50
// Deseja adicionar outro produto? (s/n): s
// Produto 2
// Nome: Lápis Preto
// Preço: 1.20
// Deseja adicionar outro produto? (s/n): n

// Lista de Produtos:
// Produto: Caneta Azul - Preço: R$ 2.50
// Produto: Lápis Preto - Preço: R$ 1.20

File: produto.cpp
#include <iostream>   // Para entrada e saída padrão
#include <vector>     // Para vetor dinâmico
#include <string>     // Para manipulação de strings
#include <iomanip>    // Para setprecision

using namespace std;

// Estrutura que representa um produto
struct Produto {
    string nome;
    float preco;
};

int main() {
    vector<Produto> produtos; // Vetor dinâmico de produtos
    char continuar;

    do {
        Produto p; // Novo produto a ser cadastrado

        cout << "Produto " << produtos.size() + 1 << endl;

        cin.ignore(); // Limpa o buffer antes de ler o nome
        cout << "Nome: ";
        getline(cin, p.nome); // Lê o nome do produto (com espaços)

        cout << "Preço: ";
        cin >> p.preco; // Lê o preço

        produtos.push_back(p); // Adiciona o produto ao vetor

        cout << "Deseja adicionar outro produto? (s/n): ";
        cin >> continuar; // Lê a resposta do usuário

    } while (continuar == 's' || continuar == 'S'); // Repete enquanto o usuário quiser

    // Exibe todos os produtos cadastrados
    cout << "\nLista de Produtos:\n";
    for (const auto& produto : produtos) {
        cout << "Produto: " << produto.nome << " - Preço: R$ "
             << fixed << setprecision(2) << produto.preco << endl;
    }

    return 0;
}

// Produto 1
// Nome: Refrigerante Laranja
// Preço: 5.50
// Deseja adicionar outro produto? (s/n): s
// Produto 2
// Nome: Salgadinho de Queijo
// Preço: 3.75
// Deseja adicionar outro produto? (s/n): s
// Produto 3
// Nome: Chocolate Amargo 70%
// Preço: 7.90
// Deseja adicionar outro produto? (s/n): n

// Lista de Produtos:
// Produto: Refrigerante Laranja - Preço: R$ 5.50
// Produto: Salgadinho de Queijo - Preço: R$ 3.75
// Produto: Chocolate Amargo 70% - Preço: R$ 7.90